src/openeo_gfmap/utils/catalogue.py

```diff
@@ -211,26 +215,46 @@ def s1_area_per_orbitstate_vvvh(
     spatial_extent = box(*bounds)
 
     # Computes if there is the full overlap for each of those states
-    union_ascending = unary_union(ascending_products)
-    union_descending = unary_union(descending_products)
+    union_ascending = unary_union(ascending_products[0])
+    union_descending = unary_union(descending_products[0])
 
     ascending_covers = union_ascending.contains(spatial_extent)
     descending_covers = union_descending.contains(spatial_extent)
 
+    # Computes max temporal gap. Include requested start and end date so we dont miss
```

Comment:
to make the code more modular I would seperate this functionality from s1_area_per_orbitstate_vvvh and call it in as a function. 

Now this code does a lot of things, instead of a single thing. 

This makes it impractical to have unit-tests in place

Reply:
Not a bad suggestion, but for me personally out of scope, because I think it requires rewriting the whole thing. You do want to avoid making the (time-consuming) catalogue queries multiple times (like once for the geometries and once for the timestamps) so also those have to go out of this method then. Could indeed increase the readability of the code, but I'm afraid we'll have to find someone to do that part as I just wanted to "fix a bug".

Reply:
I understand the need for a quick fix, but I believe we should prioritize long-term maintainability over short-term solutions. I agree that it could take more time upfront, but I think it would save time in the long run by making the code more testable and maintainable.

Reply:
I moved the computation of the maximum temporal gap in a separate function and added an unit test for it